pkg/cli/mirror/publish.go

```diff
@@ -452,16 +455,95 @@ func (o *MirrorOptions) fetchBlob(ctx context.Context, regctx *registryclient.Co
 }
 
 func unpack(archiveFilePath, dest string, filesInArchive map[string]string) error {
-	archivePath, found := filesInArchive[archiveFilePath]
-	if !found {
+	archivePaths := getArchivePathsFromMap(filesInArchive, archiveFilePath)
+
+	if len(archivePaths) == 0 {
 		return &ErrArchiveFileNotFound{archiveFilePath}
 	}
-	if err := archive.NewArchiver().Extract(archivePath, archiveFilePath, dest); err != nil {
-		return err
+
+	for _, archivePath := range archivePaths {
+		if archiveFilePath == config.CatalogsDir {
+			err := unarchive(archivePath, dest)
+			if err != nil {
+				klog.V(2).Info("unarchive failed with the error: %s", err.Error())
+				return err
+			}
+		} else {
+			if err := archive.NewArchiver().Extract(archivePath, archiveFilePath, dest); err != nil {
+				return err
+			}
+			if _, err := os.Stat(filepath.Join(dest, archiveFilePath)); err != nil {
+				return err
+			}
+		}
 	}
-	if _, err := os.Stat(filepath.Join(dest, archiveFilePath)); err != nil {
-		return err
+
+	return nil
+}
+
+func getArchivePathsFromMap(m map[string]string, key string) []string {
+	var archivePaths []string
+	for k, v := range m {
+		if strings.Contains(k, key) {
+			if slices.Contains(archivePaths, v) {
+				continue
+			}
+			archivePaths = append(archivePaths, v)
+		}
 	}
+	return archivePaths
+}
+
+func unarchive(archivePath, dest string) error {
```

Comment:
@aguidirh - it would be good to have some unit testing, but if its going to take too long we can live without it

Reply:
Hi @lmzuccarelli,

As we already discussed, since v1 is going to be replaced by v2, developing these tests are not going to bring value for oc-mirror so we are not going to develop them.